Sources/AbortExtended.swift

```diff
@@ -1,63 +1,51 @@
 import Vapor
 import HTTP
 
-public enum AbortExtended {
-    case badRequest(metadata: Node)
-    case notFound(metadata: Node)
-    case serverError(metadata: Node)
-    case custom(status: Status, message: String, metadata: Node)
-}
+/// Customized error type.
+public struct AbortExtended: AbortError {
+    public let message: String
+    public let code: Int
+    public let status: Status
+    public let metadata: Node?
 
-extension AbortExtended: AbortError {
-    public var message: String {
-        switch self {
-        case .badRequest:
-            return Abort.badRequest.message
-        case .notFound:
-            return Abort.notFound.message
-        case .serverError:
-            return Abort.serverError.message
-        case .custom(status: _, message: let message, metadata: _):
-            return message
-        }
-    }
-    
-    public var code: Int {
-        switch self {
-        case .badRequest:
-            return Abort.badRequest.code
-        case .notFound:
-            return Abort.notFound.code
-        case .serverError:
-            return Abort.serverError.code
-        case .custom(status: let status, message: _, metadata: _):
-            return status.statusCode
-        }
+    /// Creates a customized error with the given values.
+    ///
+    /// - Parameters:
+    ///   - status: The HTTP status code to return. 
+    ///     Defaults to `Status.internalServerError`.
+    ///   - code: An integer representation of the error. 
+    ///     Defaults to `Status.internalServerError.statusCode`.
+    ///   - message: Textual representation on the error. 
+    ///     Defaults to `Status.internalServerError.reasonPhrase`.
+    ///   - metadata: Custom metadata. Defaults to `nil`.
+    ///   - report: Indicates if middleware(s) should report this error. 
+    ///     Defaults to `true`.
+    ///     This will set `metadata["report"]`.
+    /// - Returns: Customized error (conforming to `AbortError`).
+    public static func custom(
+        status: Status = .internalServerError,
+        code: Int = Status.internalServerError.statusCode,
+        message: String = Status.internalServerError.reasonPhrase,
+        metadata: Node? = nil,
+        report: Bool = true
+    ) -> AbortExtended {
+        return AbortExtended(
+            message: message,
+            code: code,
+            status: status,
+            metadata: AbortExtended.resolveMetadata(metadata, report: report)
+        )
     }
-    
-    public var status: Status {
-        switch self {
-        case .badRequest:
-            return Abort.badRequest.status
```

Comment:
For these, you should be able to just return the base Status cases. [See](https://github.com/vapor/engine/blob/1f95094ee470408309e98dd56b2251210d6a2a3d/Sources/HTTP/Models/Status/Status.swift#L48)

```swift
return Status.badRequest
```